dataset_builder/humaneval_to_kt.py

```diff
@@ -0,0 +1,179 @@
+# This script translates problems from the OpenAI HumanEval dataset into TypeScript.
+import re
+import ast
+from typing import List
+from base_language_translator import LanguageTranslator
+
+DOCSTRING_LINESTART_RE = re.compile("""\n(\s+)""")
+
+needs_hashmap = False
+
+def to_camel_case(snake_str):
+    components = snake_str.split('_')
+    # We capitalize the first letter of each component except the first one
+    # with the 'title' method and join them together.
+    return components[0] + ''.join(x.title() for x in components[1:])
+
+def translate_type(t):
+    global needs_hashmap
+    match t:
+        case ast.Subscript(ast.Name(id), slice, ctx):
+            match id:
+                case "List":
+                    return "MutableList<" + translate_type(slice) + ">"
```

Comment:
Do you have a sense of what is "usual" in Kotlin? Is there an immutable list type that is more common, e.g.., in examples?

Reply:
I believe that there is an immutable list type, which is just <List>, I didn't know if the humaneval benchmarks had any tests with mutable lists, so for safety I chose to translate List to MutableList.

Reply:
The humaneval benchmarks are agnostic to whether the lists are mutable or immutable. They work with either, so its just  a matter of which is more common in Kotlin.